<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1920</width>
    <height>1080</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1920</width>
    <height>1080</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>641</width>
      <height>441</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QTableView" name="UTableViw"/>
     </item>
     <item>
      <widget class="QPushButton" name="unique_ptr">
       <property name="text">
        <string>unique_ptr</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>631</width>
      <height>441</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QTableView" name="STableView"/>
     </item>
     <item>
      <widget class="QPushButton" name="shared_ptr">
       <property name="text">
        <string>shared_ptr</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>10</y>
      <width>651</width>
      <height>441</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QTableView" name="WEakTablView"/>
     </item>
     <item>
      <widget class="QPushButton" name="weak_ptr">
       <property name="text">
        <string>weak_ptr</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>460</y>
      <width>211</width>
      <height>149</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="1" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="Uni_swap">
         <property name="text">
          <string>Swap</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="U_sw_1"/>
       </item>
       <item>
        <widget class="QSpinBox" name="U_sw_2"/>
       </item>
      </layout>
     </item>
     <item row="0" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="Uni_release">
         <property name="text">
          <string>Release</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="u_release"/>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="Uni_reset">
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="u_reset"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>460</y>
      <width>201</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="1" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QPushButton" name="Sha_swap">
         <property name="text">
          <string>Swap</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sha_sw"/>
       </item>
       <item>
        <widget class="QSpinBox" name="sha_2sw"/>
       </item>
      </layout>
     </item>
     <item row="0" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QPushButton" name="Sha_reset">
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sha_reset"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>460</y>
      <width>201</width>
      <height>109</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QPushButton" name="We_reset">
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="we_res"/>
       </item>
      </layout>
     </item>
     <item row="1" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QPushButton" name="We_lock">
         <property name="text">
          <string>Lock</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="we_lo"/>
       </item>
      </layout>
     </item>
     <item row="2" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QPushButton" name="We_swap">
         <property name="text">
          <string>Swap</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="we_sw1"/>
       </item>
       <item>
        <widget class="QSpinBox" name="we_sw2"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QSpinBox" name="U_Move_1">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>640</y>
      <width>57</width>
      <height>29</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="U_Move_2">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>640</y>
      <width>57</width>
      <height>29</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="Uni_move">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>640</y>
      <width>100</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Move</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1920</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
